File: hw/ramos/u2/hw25MongoDBConnection/java/ContactstBook/src/main/java/ec/edu/espe/contactsbook/controller/ContactDAO.java
package ec.edu.espe.contactsbook.controller;

/**
 *
 * @author Paulo Ramos
 */
import com.mongodb.client.MongoCollection;
import ec.edu.espe.contactsbook.model.Contact;
import org.bson.Document;

public class ContactDAO {

    public boolean save(Contact Contact) {

        try {
            MongoCollection<Document> collection = MongoConnection.getConnection().getCollection("Contacts");

            Document doc = new Document("First Name", Contact.getFirstName())
                    .append("Last Name", Contact.getLastName())
                    .append("Birth Date", Contact.getBirthDate())
                    .append("Age",Contact.getAge())
                    .append("Type", Contact.getTypeOfContact())
                    .append("Sex", Contact.getSex())
                    .append("Hobbies", Contact.getHobbies())
                    .append("Comments", Contact.getComments());

            collection.insertOne(doc);
            return true;
        } catch (Exception e) {
            System.out.println("Error MongoDB:" + e.getMessage());
            return false;
        }
    }
}

File: hw/ramos/u2/hw25MongoDBConnection/python/controller/ContactDAO.py
# contact_dao.py
from controller.MongoConnection import MongoConnection
from datetime import datetime, date


class ContactDAO:

    @staticmethod
    def fix_date(contact_dict):
        """Convierte datetime.date a datetime.datetime para MongoDB."""
        birth = contact_dict.get("Birth Date")

        if isinstance(birth, date):  
            contact_dict["Birth Date"] = datetime(
                birth.year,
                birth.month,
                birth.day
            )

        return contact_dict

    def save(self, contact):
        try:
            db = MongoConnection.get_connection()
            collection = db["Contacts"]

            # Convertimos el contact a diccionario
            data = contact.to_dict()

            # Convertimos fecha incompatible → compatible
            data = ContactDAO.fix_date(data)

            # Insertamos en Mongo
            collection.insert_one(data)
            return True

        except Exception as e:
            print("Error MongoDB:", e)
            return False
